### PR TITLE
Add custom widget for setting row with ToggleButtons

### PR DESCRIPTION
**Pull request by mivoligo**

### CODE REVIEW

File: lib/view/widgets/toggle_buttons_setting_row.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  final String actionLabel;
+  final String? actionDescription;
+  final List<String> labels;
+  final List<bool>? selectedValues;
+  final Function(int) onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    final _selectedValues = selectedValues;
```
Comment (jpnurmi) [inline lib/view/widgets/toggle_buttons_setting_row.dart:22]: It's a matter of taste and I'm not sure what's Frederik's preference for this, but alternatively, the extra local variable could be dropped and `isSelected: _selectedValues!` used below

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

**Comment by Feichtmeier** [comment]
Ah that's a nice idea! Thanks 😌👍

File: lib/view/widgets/toggle_buttons_setting_row.dart (hunk 1 above)
Comment (Feichtmeier) [reply lib/view/widgets/toggle_buttons_setting_row.dart:22]: Woops, did not see this. Looking ong GitHub mobile app is not always a good idea 😢 Sorry!
Eventually we can look at this in a clean up PR in the future

Comment (mivoligo) [reply lib/view/widgets/toggle_buttons_setting_row.dart:22]: I used type promotion on null checks in other custom widgets. For me it looks a bit _cleaner_ without the `!`. To be honest I don't know if there's any significant difference between those ways of doing this. If it's just a matter of taste, as you say, we should change that in other widgets if we decide one way or the other.

**Pull request merged**